### PR TITLE
Synchronous Genesis

### PR DESCRIPTION
**Pull request by maackle**
During genesis, a new `genesis_self_check` callback gets run instead of the usual validation functions, so that genesis can be fully performed while installing an app. If the self check fails, app installation fails. If it succeeds, the app is installed and ready to activate.

The self check does not fully guarantee that your source chain is valid, because other agents may perform network checks upon your joining which you were not able to perform yourself. If no network access is necessary to validate the genesis elements though, the the self-check is equivalent to the normal validation. If network access is necessary, then the self-check is a best-effort check to screen for as many errors as possible before joining the network.

### CODE REVIEW

File: crates/holochain/src/conductor/conductor.rs [hunk 1]
```diff
@@ -433,22 +443,29 @@ where
...
+                    let env = EnvironmentWrite::new(
+                        &root_env_dir,
+                        EnvironmentKind::Cell(cell_id_inner.clone()),
+                        keystore.clone(),
+                    )?;
+                    Cell::genesis(cell_id_inner, conductor_handle, env, ribosome, proof).await
+                })
+                .map_err(CellError::from)
+                .and_then(|result| async move { result.map(|_| cell_id) })
+                .await
```
Comment (freesig) [inline crates/holochain/src/conductor/conductor.rs:467]: This seems to we spawning and then immediately awaiting. Perhaps you just want to await directly without the spawn?

File: crates/holochain/src/core/workflow/genesis_workflow.rs [hunk 2]
```diff
@@ -44,21 +56,41 @@ pub async fn genesis_workflow<'env, Api: CellConductorApiT>(
...
 
     if workspace.source_chain.has_genesis() {
         return Ok(());
     }
 
+    let result = ribosome.run_genesis_self_check(
+        GenesisSelfCheckHostAccess,
+        GenesisSelfCheckInvocation {
+            payload: GenesisSelfCheckData {
+                // dna_def: dna_file.dna_def(),
```
Comment (freesig) [inline crates/holochain/src/core/workflow/genesis_workflow.rs:82]: remove?
```suggestion
```

**Review by freesig — APPROVED** [review]
Looks good to me, just a minor suggestion about spawning

File: crates/holochain/src/conductor/conductor.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/conductor/conductor.rs:467]: This is a collection of tasks that gets executed in parallel later. Also I didn't write this in this PR.

**Pull request merged**